presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -696,4 +728,277 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+    }
+
+    public Optional<TableToken> getTableToken(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, table.getSchemaTableName());
+        return Optional.ofNullable(icebergTable.currentSnapshot())
+            .map(snapshot -> new TableToken(snapshot.snapshotId()));
+    }
+
+    public boolean isTableCurrent(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<TableToken> tableToken)
+    {
+        IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        Optional<TableToken> currentToken = getTableToken(session, handle);
+
+        if (!tableToken.isPresent() || !currentToken.isPresent()) {
+            return false;
+        }
+
+        return tableToken.get().getSnapshotId() == currentToken.get().getSnapshotId();
+    }
+
+    @Override
+    public MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession session, SchemaTableName materializedViewName)
+    {
+        Map<String, Optional<TableToken>> refreshStateMap = getMaterializedViewToken(session, materializedViewName);
+        if (refreshStateMap.isEmpty()) {
+            return new MaterializedViewFreshness(false);
+        }
+
+        for (Map.Entry<String, Optional<TableToken>> entry : refreshStateMap.entrySet()) {
+            List<String> strings = Splitter.on(".").splitToList(entry.getKey());
+            checkState(strings.size() >= 2);
+            if (strings.size() == 3) {
+                strings = strings.subList(1, 3);
+            }
+            String schema = strings.get(0);
+            String name = strings.get(1);
+            SchemaTableName schemaTableName = new SchemaTableName(schema, name);
+            if (!isTableCurrent(session, getTableHandle(session, schemaTableName), entry.getValue())) {
+                return new MaterializedViewFreshness(false);
+            }
+        }
+        return new MaterializedViewFreshness(true);
+    }
+
+    private Map<String, Optional<TableToken>> getMaterializedViewToken(ConnectorSession session, SchemaTableName name)
+    {
+        Map<String, Optional<TableToken>> viewToken = new HashMap<>();
+        Optional<ConnectorMaterializedViewDefinition> materializedViewDefinition = getMaterializedView(session, name);
+        if (!materializedViewDefinition.isPresent()) {
+            return viewToken;
+        }
+
+        String storageTableName = materializedViewDefinition.get().getProperties().getOrDefault(STORAGE_TABLE, "").toString();
+        org.apache.iceberg.Table icebergTable = getIcebergTable(metastore, hdfsEnvironment, session, new SchemaTableName(name.getSchemaName(), storageTableName));
+        String dependsOnTables = icebergTable.currentSnapshot().summary().getOrDefault(DEPENDS_ON_TABLES, "");
+        if (!dependsOnTables.isEmpty()) {
```

Comment:
I don't think we need to check for empty, since splitter on empty will return an empty map.

Reply:
Splitting an empty string on comma yields an empty string. But empty string does not qualify as a valid 'key-value' pair with separator '='. The 'MapSplitter' class says:
`* @throws IllegalArgumentException if the specified sequence does not split into valid map
     *     entries, or if there are duplicate keys`
So, when dependsOnTables is empty, it results in the following error. Hence we need that `if (!dependsOnTables.isEmpty()) {` check.
`Query 20201030_182956_00010_ttinc failed: Chunk [] is not a valid entry
java.lang.IllegalArgumentException: Chunk [] is not a valid entry
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:217)
	at com.google.common.base.Splitter$MapSplitter.split(Splitter.java:526)
	at io.prestosql.plugin.iceberg.IcebergMetadata.getMaterializedViewToken(IcebergMetadata.java:997)
	at io.prestosql.plugin.iceberg.IcebergMetadata.getMaterializedViewFreshness(IcebergMetadata.java:965)`